pkg/controller/apiextensions/composite/reconciler.go

```diff
@@ -430,17 +432,15 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 		r.record.Event(cr, event.Normal(reasonPublish, "Successfully published connection details"))
 	}
 
-	// TODO(muvaf): Report which resources are not ready.
-
-	// TODO(muvaf): If a resource becomes Unavailable at some point, should we still
-	// report it as Creating?
-	wait := longWait
-	cr.SetConditions(runtimev1alpha1.Available())
```

Comment:
Ah was this causing needless reconciles when not ready? I ran into this when implementing package manager as well

Reply:
Yeah, exactly. It was causing us to update the `lastTransitionTime` of the condition on every reconcile.

Reply:
Nice catch 👍